client/src/components/Auth/Register/Register.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter, Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import { registerUser } from '../../../actions/authActions';
+import { Container, Header, TopText, ContentArea, Footer } from '../Auth.styled';
+import FormGroup from '../../Shared/Forms/FormGroup';
+import { Button } from '../../Shared/Forms/Form.styled';
+
+class Register extends Component {
+	state = {
+		name: '',
+		email: '',
+		password: '',
+		password2: '',
+		errors: {},
+	};
+
+	static propTypes = {
+		registerUser: PropTypes.func.isRequired,
+		errors: PropTypes.shape({
+			name: PropTypes.string,
+			email: PropTypes.string,
+			password: PropTypes.string,
+			password2: PropTypes.string,
+		}).isRequired,
+		auth: PropTypes.shape({
+			isAuthenticated: PropTypes.bool.isRequired,
+			user: PropTypes.object.isRequired,
+		}).isRequired,
+	};
+
+	componentWillMount() {
+		const { auth, history } = this.props;
+		if (auth.isAuthenticated) history.push('/dashboard');
+	}
+
+	componentWillReceiveProps(nextProps) {
+		if (nextProps.errors) {
+			this.setState({ errors: nextProps.errors });
+		}
+	}
+
+	handleChange = e => {
+		const { name, value } = e.target;
+		this.setState({ [name]: value });
+	};
+
+	handleSubmit = e => {
+		e.preventDefault();
+		// eslint-disable-next-line no-shadow
+		const { registerUser, history } = this.props;
+		const { name, email, password, password2 } = this.state;
```

Comment:
Quick note: Just to point out that you can use rest operator to gather values from the state here:

```
const { errors, ...newUser } = this.state;
registerUser(newUser, history);
```

Reply:
Nice! 😀